### ISSUE TITLE
Yield Sig Figs not updating

### ISSUE THREAD

**Issue by riversn** [bradmartin333/XferSuite#256]
Significant figures for yield doesn't change when user changes the settings.

Maybe it has to be set during the initial parsing/opening the file and not after evaluating it?

![image](https://github.com/bradmartin333/XferSuite/assets/101815602/b1a4a51f-ab02-4332-8984-ea5db5ba44d2)

**Comment by bradmartin333** [comment]
yes! bug!!! I round correctly, but format the string incorrectly

**Comment by bradmartin333** [comment]
you still have to plot again after changing, like most settings

**Comment by bradmartin333** [comment]
fix: `int numDecimalPlaces = Math.Max(0, sigFigs - (int)(percentage / 10 + 1));`

**Comment by riversn** [comment]
Awesome! Found this one because we were getting such good yields lately :)

**Comment by bradmartin333** [comment]
lol okay - can you send me the file so I can make sure - I have never had a file with that many 9s

**Comment by riversn** [comment]
Not really, it's production data... Just make a seyrup with a prime number yield

**Comment by riversn** [comment]
I mean... rounding is good - but it should follow the sig fig user determined amount.

**Comment by bradmartin333** [comment]
and you did click plot again?

**Comment by riversn** [comment]
Yep

**Comment by bradmartin333** [comment]
changing to

```cs
double percentage = Math.Round(pass / total, sigFigs) * 100;
int numDigitsBeforeDecimal = (int)Math.Log10(percentage) + 1;
int numDecimalPlaces = Math.Max(0, sigFigs - numDigitsBeforeDecimal);
```

**Comment by riversn** [comment]
Looks fixed now!

**Comment by bradmartin333** [comment]
ok - so sig figs are good

I use [this](https://learn.microsoft.com/en-us/dotnet/api/system.math.round?view=netframework-4.7.2) for rounding... there are some options if you want me to customize it
